onecodex/analyses.py

```diff
@@ -536,6 +490,63 @@ def _get_highest_unclassified_tax_id(self, tax_id, level):
 
         return highest_tax_id
 
+    @staticmethod
+    def _make_labels_by_item_id(metadata, label):
+        """Make labels from
+
+        Parameters
+        ----------
+        metadata : `pandas.DataFrame`
+        label : `str` or `callable`
+
+        Returns
+        -------
+        `dict`
+            Keys are from metadata.index. Values are generated labels.
+        """
+        import pandas as pd
+
+        raw_result = {}
+
+        if isinstance(label, str):
+            if label in metadata.columns:
+                raw_result = dict(metadata[label].items())
+            else:
+                raise OneCodexException(
+                    "Label field {} not found. Choose from: {}".format(
+                        label, ", ".join(metadata.keys())
+                    )
+                )
+        elif callable(label):
+            for item_id, item_meta in metadata.to_dict(orient="index").items():
+                item_label = label(item_meta)
+                if not isinstance(item_label, six.string_types):
```

Comment:
Since `six.string_types` was already replaced with `str` a few lines up, should we do the same here?